sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ProtoDescriptorGenerator.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl
+
+import java.lang.reflect.Method
+import java.lang.reflect.ParameterizedType
+
+import scala.jdk.CollectionConverters.IterableHasAsJava
+import scala.jdk.CollectionConverters.IterableHasAsScala
+
+import kalix.javasdk.action.Action
+import kalix.javasdk.valueentity.ValueEntity
+import com.fasterxml.jackson.dataformat.protobuf.ProtobufMapper
+import com.fasterxml.jackson.dataformat.protobuf.schema.ProtobufSchema
+import com.google.protobuf.DescriptorProtos
+import com.google.protobuf.Descriptors
+import com.fasterxml.jackson.dataformat.protobuf.schema.FieldType
+import com.fasterxml.jackson.dataformat.protobuf.schema.FieldType._
+
+object ProtoDescriptorGenerator {
+
+  sealed trait TypesSelector {
+    def returnParamTypeName(method: Method) = {
+      val genericReturnType = method.getGenericReturnType.asInstanceOf[ParameterizedType]
+      genericReturnType.getActualTypeArguments()(0).getTypeName
+    }
+
+    def returnParamType(method: Method) =
+      try {
+        // FIXME: don't load class, but extract the simple name using regex
+        Class.forName(returnParamTypeName(method))
+      } catch {
+        case e: ClassNotFoundException =>
+          throw new RuntimeException(e)
+      }
+
+    def inputType: Class[_]
+    def outputType: Class[_]
+    def allMessageTypes: Set[Class[_]]
+  }
+
+  case class ActionTypesSelector(method: Method) extends TypesSelector {
+    override val inputType: Class[_] = method.getParameterTypes()(0)
+    override val outputType: Class[_] = returnParamType(method)
+    override def allMessageTypes: Set[Class[_]] =
+      Set(outputType, inputType)
+  }
+
+  case class ValueEntityTypesSelector(method: Method) extends TypesSelector {
+    // gRPC input is the second param
+    override val inputType: Class[_] = method.getParameterTypes()(1)
+    override val outputType: Class[_] = returnParamType(method)
+
+    override def allMessageTypes: Set[Class[_]] =
+      Set(outputType, inputType, method.getParameterTypes()(0))
+  }
+
+  private object InternalGenerator {
+
+    private def mapScalarType(fieldType: FieldType): DescriptorProtos.FieldDescriptorProto.Type =
```

Comment:
I guess we won't be needing all this if we anyway always work with json in anys?

Reply:
we will need to generate schema for types that will be used in views.